great_expectations/rule_based_profiler/data_assistant/data_assistant_runner.py

```diff
@@ -105,12 +50,140 @@ def __init__(
         self._data_assistant_cls = data_assistant_cls
         self._data_context = data_context
 
+        self._profiler = self.get_profiler()
+
+        setattr(self, "run", self.run_impl())
```

Comment:
by the time we get to `run`, the signature is configured in the way that each `DataAssistant` requires